src/attributes/classes.js

```diff
@@ -24,19 +22,17 @@ jQuery.fn.extend({
 			});
 		}
 
-		if ( proceed ) {
-			// The disjunction here is for better compressibility (see removeClass)
-			classes = ( value || "" ).match( rnotwhite ) || [];
+		if ( typeof value === "string" && value ) {
+			classes = value.match( rnotwhite ) || [];
 
-			for ( ; i < len; i++ ) {
-				elem = this[ i ];
+			while ( ( elem = this[ i++ ] ) ) {
```

Comment:
Of course, scrolling down just two lines reveals that **this** change actually _doesn't_ affect behavior. Still, there are analogous places in the codebase where that doesn't hold, and my question stands.